src/mmrelay/migrate.py

```
@@ -1581,8 +1582,11 @@ def migrate_database(
         ):
             try:
                 db_uri = f"{main_db_staged.resolve().as_uri()}?mode=ro"
-                with sqlite3.connect(db_uri, uri=True) as conn:
-                    result = conn.execute("PRAGMA integrity_check").fetchone()
+                with contextlib.closing(sqlite3.connect(db_uri, uri=True)) as conn:
+                    with conn as managed_conn:
```

Ruff (0.15.10)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)